Wikipedia/Code/WelcomePageViewController.swift

```diff
@@ -0,0 +1,199 @@
+        skipButton.setTitle(CommonStrings.skipTitle, for: .normal)
+        skipButton.titleLabel?.numberOfLines = 1
+        skipButton.addTarget(self, action: #selector(skip(_:)), for: .touchUpInside)
+        view.addSubview(skipButton)
+        let heightConstraint = skipButton.heightAnchor.constraint(equalToConstant: buttonHeight)
+        let bottomConstraint = skipButton.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        let leadingConstraint = skipButton.leadingAnchor.constraint(greaterThanOrEqualTo: view.leadingAnchor, constant: buttonHorizontalSpacing)
+        let trailingConstraint = skipButton.trailingAnchor.constraint(lessThanOrEqualTo: view.centerXAnchor, constant: -buttonXOffset)
+        NSLayoutConstraint.activate([
+            heightConstraint,
+            bottomConstraint,
+            leadingConstraint,
+            trailingConstraint
+        ])
+    }
+
+    @objc private func next(_ sender: UIButton) {
+        guard
+            let visibleViewController = visibleViewController,
+            let nextViewController = viewController(after: visibleViewController)
+        else {
+            return
+        }
+        view.isUserInteractionEnabled = false
+        animatePageControlButtons(for: nextViewController)
+        setViewControllers([nextViewController], direction: direction, animated: true) { _ in
+            self.view.isUserInteractionEnabled = true
+        }
+    }
+
+    @objc private func skip(_ sender: UIButton) {
+        dismiss(animated: true)
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        let buttonFont = UIFont.wmf_font(.semiboldFootnote, compatibleWithTraitCollection: traitCollection)
+        nextButton.titleLabel?.font = buttonFont
+        skipButton.titleLabel?.font = buttonFont
+    }
+
+    private var direction: UIPageViewController.NavigationDirection {
+        return UIApplication.shared.wmf_isRTL ? .reverse : .forward
+    }
+
+    private func viewController(at index: Int) -> UIViewController? {
+        guard index >= 0, allViewControllers.count > index else {
+            return nil
+        }
+        return allViewControllers[index]
+    }
+
+    private var visibleViewController: UIViewController? {
+        return viewControllers?.first
+    }
+}
+
+extension WelcomePageViewController: Themeable {
+    func apply(theme: Theme) {
+        guard viewIfLoaded != nil else {
+            self.theme = theme
+            return
+        }
+        allViewControllers.forEach { ($0 as? Themeable)?.apply(theme: theme) }
+        view.backgroundColor = theme.colors.midBackground
+        pageControl?.pageIndicatorTintColor = theme.colors.pageIndicator
+        pageControl?.currentPageIndicatorTintColor = theme.colors.pageIndicatorCurrent
+        nextButton.tintColor = theme.colors.link
+        skipButton.tintColor = UIColor(0xA2A9B1)
```

Comment:
maybe this should be a theme color?

Reply:
good point 👍

Reply:
although maybe when we consolidate all the onboarding stuff, we could get away with just having one `public static wmf_someShadeOfGray` since this button has the same colors in all the themes right now 🧐